File: 2022/09/12/arrayPartition.js
/**
 * https://leetcode.com/problems/array-partition/
 * @author Andrew Jarombek
 * @since 9/12/2022
 */

const arrayPairSum = function(nums) {
    nums.sort((a, b) => a - b);
    return nums.reduce((acc, val, index) => index % 2 === 0 ? acc + val : acc);
};

console.assert(arrayPairSum([1, 2, 3, 4]) === 4);

File: 2023/07/16/go/array_partition.go
/**
 * https://leetcode.com/problems/array-partition/
 * Author: Andrew Jarombek
 * Date: 7/16/2023
 */

package _go

import "sort"

func arrayPairSum(nums []int) int {
	sort.Slice(nums, func(i, j int) bool {
		return nums[i] > nums[j]
	})

	result := 0

	for i := 1; i < len(nums); i += 2 {
		result += nums[i]
	}

	return result
}
